tests/rados/customer_scenarios.py

```diff
@@ -0,0 +1,385 @@
+    if (
+        final_commmits["osdmap_first_committed"]
+        < init_commmits["osdmap_first_committed"]
+    ):
+        log.error(
+            f"The OSD map has not been updated of first commits\n"
+            f"The commits are initial commits : {init_commmits}, final commits : {final_commmits}"
+        )
+        return False
+
+    # Final updates need to be more than initial updates as there are OSD map changes during the duration of test
+    if (
+        final_commmits["osdmap_last_committed"]
+        <= init_commmits["osdmap_last_committed"]
+    ):
+        log.error(
+            f"The OSD map has not been updated of last commits\n"
+            f"The commits are initial commits : {init_commmits}, final commits : {final_commmits}"
+        )
+        return False
+
+    # The number of OSD mappings present on cluster should not exceed 900 in total
+    if (
+        final_commmits["osdmap_last_committed"]
+        - final_commmits["osdmap_first_committed"]
+    ) > 900:
+        log.error(
+            f"There are still too many old commits in Mon DB. OSD map not trimmed as per needed\n"
+            f"The commits are initial commits : {init_commmits}, final commits : {final_commmits}"
+        )
+        return False
+
+    log.info("mon DB was trimmed successfully")
+    return True
+
+
+def get_mondb_size(mon_node, mon_daemons) -> int:
+    """
+    Executes du -sch to get the size of the mon DB present at /var/lib/ceph/mon/ceph-$(hostname -s)/store.db
+    Args:
+       mon_node: Node where the mon daemon is running
+       mon_daemons: Details of all mon daemons running on the cluster
+
+    Returns: Returns a dictionary whose keys are the daemons and the values are a named tuple, containing the info.
+
+    """
+    daemon = [
+        mon for mon in mon_daemons if re.search(mon["hostname"], mon_node.hostname)
+    ][0]
+    cmd = f"podman exec {daemon['container_id']} du -ch /var/lib/ceph/mon/"
+    log.info(
+        f"Collecting the size of the DB on node: {mon_node.hostname} by executing the command : {cmd}"
+    )
+    out, err = mon_node.exec_command(sudo=True, cmd=cmd)
+    output = out.read().decode()
+    regex = r"\s*([\d]*)[M|G]\s+[\w\W]*store.db"
+    match = re.search(regex, output)
+    size = match.groups()[0] if match else 0
+    log.debug(f"the size of the cluster DB is {int(size)}")
+    return int(size)
+
+
+def reweight_crush_items(node: CephAdmin, **kwargs) -> bool:
+    """
+    Performs Re-weight of various CRUSH items, based on key-value pairs sent
+    Args:
+        node: Cephadm node where the commands need to be executed
+        **kwargs: Arguments for the commands
+
+    Returns: True -> pass, False -> fail
+
+    """
+    if kwargs.get("name"):
+        name = kwargs["name"]
+        weight = kwargs["weight"]
+        cmd = f"ceph osd crush reweight {name} {weight}"
+        node.shell([cmd])
+        return True
```

Comment:
Not returning false as I am not verifying the result of this. ( PG would be redistributed based whether we increase or decrease the weight. ) PG movement is slow and will take time based on no of objects. 

Need to implement this check. So for now assuming it worked and returning true. ( Customer faced issue where this command had no effect. No PG movement observed )

Included this in this workflow as rather than a separate one as even in this bug, we need to simulate PG movement for DB change